grakn-graph/src/main/java/ai/grakn/factory/SystemKeyspace.java

```diff
@@ -82,10 +83,11 @@
     protected static final Logger LOG = LoggerFactory.getLogger(SystemKeyspace.class);
 
     private static final ConcurrentHashMap<String, Boolean> openSpaces = new ConcurrentHashMap<>();
-    private static final AtomicBoolean isFactorySet = new AtomicBoolean(false);
-    private static InternalFactory factory;
+    static final AtomicBoolean factoryBeingInstantiated = new AtomicBoolean(false);
```

Comment:
Not sure if you can work around this but it would be better to maintain these as `private`. I know you did it so your mock class can access these fields but is there no other way to achieve the same result without having to change this implementation? 

Maybe override the `initialise` methods somehow? If there is no way so be it, but exposing fields for the sake of testing is a little bit bad.